suffuse/error.go

```diff
@@ -0,0 +1,70 @@
+package suffuse
+
+import (
+  "errors"
+  "reflect"
+)
+
+/** If you have a *Foo type which implements error and you pass it
+ *  to a method accepting error, when you pass a nil pointer it
+ *  does NOT test as nil. It's apparently a non-nil interface with
+ *  a nil value. That's what the second test is for.
+ */
+func IsNilError(e error)bool {
+  return e == nil || reflect.ValueOf(e).IsNil()
```

Comment:
Well, we are using exactly one custom error type that contains information. All other types of errors have (in their methods) `error` (an interface) as return type and will not exhibit that problem.

And, of course, while I am typing this I realize we are using a short circuiting `||` which makes any of my arguments invalid.

Still, in order to be 'go compliant' I think we should expose our custom error as an interface instead of a struct pointer while retaining the `reflect` check.